symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/stream/StreamService.java

```diff
@@ -185,19 +235,61 @@ public V3RoomDetail create(V3RoomAttributes roomAttributes) {
    * @return The rooms returned according to the given criteria.
    * @see <a href="https://developers.symphony.com/restapi/reference#search-rooms-v3">Search Rooms V3</a>
    */
-  public V3RoomSearchResults searchRooms(V2RoomSearchCriteria query) {
+  public V3RoomSearchResults searchRooms(@Nonnull V2RoomSearchCriteria query) {
     return executeAndRetry("searchRooms",
         () -> streamsApi.v3RoomSearchPost(authSession.getSessionToken(), query, null, null));
   }
 
+  /**
+   * Search rooms according to the specified criteria.
+   *
+   * @param query       The room searching criteria.
+   * @param pagination  The skip and limit for pagination.
+   * @return The rooms returned according to the given criteria.
+   * @see <a href="https://developers.symphony.com/restapi/reference#search-rooms-v3">Search Rooms V3</a>
+   */
+  public V3RoomSearchResults searchRooms(@Nonnull V2RoomSearchCriteria query, @Nullable PaginationAttribute pagination) {
+    if (pagination != null) {
+      return executeAndRetry("searhRooms",
+          () ->streamsApi.v3RoomSearchPost(authSession.getSessionToken(), query, pagination.skip(), pagination.limit()));
+    }
+    return searchRooms(query);
+  }
+
+  /**
+   * Search rooms and return in a {@link java.util.stream.Stream} according to the specified criteria.
+   *
+   * @param query       The room searching criteria.
+   * @return A {@link java.util.stream.Stream} of rooms returned according to the given criteria.
+   * @see <a href="https://developers.symphony.com/restapi/reference#search-rooms-v3">Search Rooms V3</a>
+   */
+  public java.util.stream.Stream<V3RoomDetail> searchRoomsInStream(@Nonnull V2RoomSearchCriteria query) {
```

Comment:
Please remove `@Nullable` on `PaginationAttribute pagination` and assume it is not null.